.github/workflows/code-health.yml

```diff
@@ -16,22 +16,34 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - name: Set up Go
-      uses: actions/setup-go@v5
+    - uses: actions/setup-go@v5
       with:
         go-version-file: 'go.mod'
+    - name: Install tools for mocks and linter
+      run: make tools
+    - name: Mock generation
+      run: mockery
+    - name: Check for uncommited files
+      run: |
+        export FILES=$(git ls-files -o -m --directory --exclude-standard --no-empty-directory)
+        export LINES=$(echo "$FILES" | awk 'NF' | wc -l)
+        if [ $LINES -ne 0 ]; then
+          echo "Detected files that need to be committed:"
+          echo "$FILES" | sed -e "s/^/  /"
+          echo ""
+          echo "Try running: mockery"
+          exit 1
+        fi
     - name: Build
       run: make build
+    - name: Linter
+      run: make lint
```

Comment:
linter here takes 2m instead of 3m and linter version is centralized in make file

Reply:
I am not sure how it was before 🤔 I don't understand the advantages of adding these changes, maybe you could help me understand here. Could you clarify the current state without your changes and what your changes are trying to achieve? Thanks!

Reply:
the main advantage about the linter change is that we'll have the version only in one place, now it's in 2 places and we need to remember every time we upgrade it:
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/GNUmakefile#L16
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/.github/workflows/code-health.yml#L72

Reply:
with this change I'm using make lint from GH action so version is only in one place

Reply:
no, sorry, this was an opportunistic change as i was changing something close to it

Reply:
FYI by moving out of the golangci lint action to the make file version you are loosing the action ability to comment on the line with the failure and you gonna have to parse the log files to figure out what broke

Reply:
@gssbzn that's fair  but we already have a pre-commit hook with linter so linter should rarely fail in GH

Reply:
Gus made an interesting comment on that. And I guess your answer seems more like "this is a duplicate" - so my point is that either it's useful having linter here (hence we need to make the results easily actionable, hence this does not seem a good idea) or it's not.

Since it's not actually related to the change you are proposing in the PR I would suggest to remove this change the things as it is. Your pros mentioned above to Andrea as well does not look that strong to me compared to Gus's comment.

Reply:
I pushed an offending commit (bypasing commit hooks) with current linter and the one in this PR.

Current linter: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7258950919/job/19775168073
```
Running [/home/runner/golangci-lint-1.55.0-linux-amd64/golangci-lint run --out-format=github-actions --timeout 10m] in [] ...
  Error: test helper function should start from tb.Helper() (thelper)
  
  Error: issues found
  Ran golangci-lint in 169508ms
``` 

Linter in this PR:  https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7259036317/job/19775411159?pr=1763
```
golangci-lint run
Error: internal/testutil/acc/skip.go:18:[6](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7259036317/job/19775411159?pr=1763#step:8:7): test helper function should start from tb.Helper() (thelper)
func SkipIfTFAccNotDefined(tb testing.TB) {
     ^
make: *** [GNUmakefile:68: lint] Error 1
Error: Process completed with exit code 2.
```

So apart from having the linter version only in one place, here we have information about the offending file and line whereas in the current approach we only see the error not where it happens.

Please clarify why you think the current situation is better, I'm not sure I understand the part "the action ability to comment on the line with the failure ".

Reply:
<img width="1698" alt="Screenshot 2023-12-19 at 13 59 33" src="https://github.com/mongodb/terraform-provider-mongodbatlas/assets/6604389/c8eb032e-ec77-43d8-8d75-4f689e7f8bed">

I am not gonna try to answer on behalf of Gus, but I just have a question: would we see this linter error on GH if we move the linter in this new place?